web/src/components/config.ts

```diff
@@ -0,0 +1,10 @@
+import * as React from 'react';
+
+export type Config = {
+  PortNaming: {
+    Enable: boolean;
+    PortNames: string[];
+  };
+};
```

Comment:
```suggestion
export type Config = {
    portNames?: Map<number, string>;
};
```

What about using an optional map here ? You define it as soon as it's enabled

Reply:
On the backend side it is a map so ... how can it be an array here? (I'm missing something)
if this type is used directly as input json type, then it would be `{[k:string]: string}` rather than `Map`